papers/Sepideh_Alassi_Bertram_Winter/PyCells_for_an_Open_Semiconductor_Industry

```diff
@@ -0,0 +1,270 @@
+
+After completing the setup step, the first structure to be generated is the polysilicon gate which has an octagonal shape drawn with a polysilicon layer (poly1) with a gap inside for drain contact. Later the gate structure is appended to the list of shapes which are connected to the gate pin. 
+
+ .. code-block:: python
+   
+  p1Gate = OctagonWithHole(layers.poly1, 
+		gateOuterBox, gateInnerBox,
+                td['p1_corn'], 0)
+  self.pins['G'].append(p1Gate)
+	
+Other structures of the transistor, such as diffusion layer, drain and source pins and the contacts of them, are generated serially in a predefined order. (see Figure :ref:`pin`)
+
+.. figure:: pins.pdf
+   :scale: 50%
+   :align: center
+   
+   Creation of the source and drain contacts. :label:`pin`
+  
+In order to increase the breakdown voltage of the device, regions with different doping concentrations have to be created at the drain terminal of the transistor. (see Figure :ref:`well`)
+
+.. figure:: wells.pdf
+   :scale: 50%
+   :align: center
+   
+   Deep and shallow p-well under the drain. :label:`well`
+
+Similarly an n-well should be added to define the device body, followed by a p-implant layer on source and drain. Furthermore, the bulk connection of the pmos20t device is accomplished by a contact ring around the device which has to make the connection to the n-type isolation well, for which n-type implant and a shallow n-well are required. (see Figure :ref:`bulk`)    
+
+.. figure:: bulk.pdf
+   :scale: 50%
+   :align: center
+   
+   Bulk contact as a ring with an n-implant and an n-well. :label:`bulk`
+
+Lastly, the overall device structures, such as definition layer for the thickness of the gate oxide, should be added to PyCell. The resulting layout for pmos20t is shown in Figure :ref:`pmos20`. 
+
+.. figure:: pmos20t.pdf
+   :scale: 40%
+   :align: center
+   
+   Final layout of a pmos20t transistor. :label:`pmos20`
+   
+Figure :ref:`pmos50` shows the layout generated for a transistor with 50V guard ring and multiplication factor of 2.
+
+.. figure:: pmos50.pdf
+   :scale: 35%
+   :align: center
+   
+   Layout with a 50V guard ring and a device multiplier of 2. :label:`pmos50`
+
+Optimization
+++++++++++++
+
+The computation time plays a crucial role in designing integrated circuits. The complexity of the designs and the amount of operations necessary to generate layouts of high voltage devices or the large devices increase the computation time significantly. For this reason, throughout this project the PyCells are developed to be in the most optimized shape possible without losing the clarity of the structure of the codes. For optimizing the codes, data aggregation and optimal use of local variables as well as iterations are taken into account. This has resulted in faster execution of PyCell codes for generating large and complex devices in comparison with execution time of PCells for identical devices.
```

Comment:
I am not sure I understand what you mean by "shape" in this context:

> For this reason, throughout this project the PyCells are developed to be in the
> most optimized shape possible without losing the clarity of the structure of the 
> codes. 

Do you mean geometric shape of the resistors, or some feature of the software design? I think this paragraph needs to be clarified to specifically address what kind of "optimal use of local variables" and "data aggregation" you implemented. As is, the text is too vague to tell either 

- What you did to optimize the code
- or how fast the code now runs.

Consider answering the following question:

What kind of execution time is typical, and what specific changes to the code were most effective in making this speedup possible.